frontend/src/components/editor-page/editor-pane/autocompletions/basic-completion.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * SPDX-FileCopyrightText: 2023 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+import type { CompletionContext, CompletionResult } from '@codemirror/autocomplete'
+
+/**
+ * Returns a {@link CompletionSource} for a basic regex-matching autocompletion suggestion.
+ *
+ * @param regexToMatch The regex to match in front of the cursor
+ * @param replace The string to insert as completion
+ * @param description An optional description to show besides the suggestion
+ * @return A function to test and perform the configured completion.
+ */
+export const basicCompletion =
+  (regexToMatch: RegExp, replace: string, description?: string) =>
```

Comment:
How about providing 2 different functions that either take a regex or a prefix? This would make prefix functions easier.